slackreact/_bot.py

```diff
@@ -69,11 +70,18 @@ async def _handle_futures(
                     level=logging.ERROR,
                 )
 
-    async def api_call(self, **data: Any) -> Dict[Any, Any]:
+    async def api_call(self, max_num_retries: int = 5, **data: Any) -> Dict[Any, Any]:
         method = data.pop("method")
         data["token"] = self.token
-        async with self.session.post(f"https://slack.com/api/{method}", data=data) as response:
-            return await response.json()
+        num_retries = 0
+        while True:
+            async with self.session.post(f"https://slack.com/api/{method}", data=data) as response:
+                ret = await response.json()
+            if ret.get('error') != 'ratelimited' or num_retries >= max_num_retries:
+                return ret
+            num_retries += 1
+            retry_after = int(response.headers.get('Retry-After', 1))
+            time.sleep(retry_after)
```

Comment:
Use asyncio.sleep, so we don't block the event loop.
```suggestion
            await asyncio.sleep(retry_after)
```